java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -188,17 +188,15 @@ class ReferenceModeStore private constructor(
     ): Int = callbacks.register(callback)
 
     override fun off(callbackToken: Int) {
-        containerStore.off(containerCallbackToken)
```

Comment:
This is absolutely the right move: we shouldn't have closed container at the very first callback `off()`.
Please clean up the use of `containerCallbackToken` which is unused after this change.

Reply:
Note that because the only signal to close Stores currently is when the last callback is registered, removing this instead of doing something like "if(callbacks.isEmpty()) containerStore.off(containerCallbackToken)" inadvertently introduced the unbounded memory leak this call fixed.

Reply:
iirc there was a bug where there are multiple and racing callback registrations and unregistrations.
So on() -> on() -> off() -> off(), here we close everything -> on(), boom!
I'm not sure if the issue still stands true given the codebase has already changed a lot.